src/main/kotlin/com/exactpro/th2/codec/fixng/FixNgCodec.kt

```diff
@@ -190,23 +192,37 @@ class FixNgCodec(dictionary: IDictionaryStructure, settings: FixNgCodecSettings)
             }
         }
 
-        // in dirty mode we'll use tag as field name if we have field duplication
-        val fieldName = if (isDirty && target.contains(name)) {
-            tag.toString()
+        if (!tagsSet.add(tag)) {
+            handleError(isDirty, context, "Duplicate $name field ($tag) with value: $value", value)
+        }
+
+        val targetMap = if (isDecodeComponentsToNestedMaps) {
+            @Suppress("UNCHECKED_CAST")
+            path.fold(target) { map, key -> map.computeIfAbsent(key) { mutableMapOf<String, Any>() } as MutableMap<String, Any> }
         } else {
-            name
+            target
         }
 
-        val previous = target.put(fieldName, decodedValue)
+        val fieldName = if (isDirty && targetMap.contains(name)) {
+            // in dirty mode we'll use tag as field name if we have field duplication
```

Comment:
Could you please revert part for duplication handling

Reply:
reverted